steel/src/host/db.rs

```diff
@@ -33,7 +33,7 @@ pub enum ProviderDbError<E: std::error::Error> {
 }
 
 /// A revm [Database] backed by a [Provider].
-pub struct ProviderDb<P: Provider> {
+pub struct ProviderDb<P> {
```

Comment:
You are absolutely right, technically these bounds are unnecessary and I am not sure what the most idiomatic way is. However, the `ProviderDb` has no other implementation and must always be used with a provider (which is also in the doc), so I thought it might make sense to keep them for clarity.

Reply:
Yeah, it doesn't right now, but even though that's the case, it causes very parasitic requirements of any types that use this to also define those boundaries, and gets messy for consumers in some cases. This just gives more flexibility, and future proofs imo. Happy to change back if you feel strongly about it, though.

Reply:
No strong feeling at all. So let's keep this change.